stellar-multitenant/pkg/tenant/manager.go

```diff
@@ -53,32 +56,68 @@ func (m *Manager) GetDSNForTenant(ctx context.Context, tenantName string) (strin
 	return router.GetDSNForTenant(dataSourceName, tenantName)
 }
 
+func (m *Manager) GetDSNForTenantByID(ctx context.Context, id string) (string, error) {
+	t, err := m.GetTenantByID(ctx, id)
+	if err != nil {
+		return "", fmt.Errorf("getting tenant for ID %s: %w", id, err)
+	}
+
+	return m.GetDSNForTenant(ctx, t.Name)
+}
+
 var selectQuery string = `
 	SELECT 
 		*
 	FROM
 		tenants t
-	%s
 `
 
 // GetAllTenants returns all tenants in the database.
-func (m *Manager) GetAllTenants(ctx context.Context) ([]Tenant, error) {
-	tnts := []Tenant{}
-
-	query := fmt.Sprintf(selectQuery, "ORDER BY t.name ASC")
+func (m *Manager) GetAllTenants(ctx context.Context, queryParams *QueryParams) ([]Tenant, error) {
+	if queryParams == nil {
+		queryParams = &QueryParams{
+			Filters: map[FilterKey]interface{}{
+				FilterKeyOutStatus: DeactivatedTenantStatus,
+			},
+			SortBy:    data.SortFieldName,
+			SortOrder: data.SortOrderASC,
+		}
+	}
 
-	err := m.db.SelectContext(ctx, &tnts, query)
+	tnts := []Tenant{}
+	query, params := m.newManagerQuery(selectQuery, queryParams)
+	err := m.db.SelectContext(ctx, &tnts, query, params...)
 	if err != nil {
 		return nil, fmt.Errorf("getting all tenants: %w", err)
 	}
 
 	return tnts, nil
 }
 
+// GetTenant is a generic method that fetches a tenant based on queryParams.
+func (m *Manager) GetTenant(ctx context.Context, queryParams *QueryParams) (*Tenant, error) {
+	var t Tenant
+	q, params := m.newManagerQuery(selectQuery, queryParams)
+	if err := m.db.GetContext(ctx, &t, q, params...); err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return nil, ErrTenantDoesNotExist
+		}
+		return nil, fmt.Errorf("getting tenant: %w", err)
+	}
+	return &t, nil
+}
+
```

Comment:
👋 💡 Can't you improve the core reusability by calling GetTenant from GetTenantByID, GetTenantByName, GetTenantByIDOrName, and GetDefault?

Reply:
done! I don't think this can be applicable to `GetDefault` though